### PR TITLE
feat(quantic): added searchhub and pipeline params to ssb

### PR DESCRIPTION
**Pull request by nathanlb**
The SSB doesn't permit to set a `searchhub` or `pipeline` to the engine via arguments. A user should be able to set those values to configure query suggestions within the SSB and send analytics to right hub.

Related [Discuss](https://discuss.coveo.com/t/qs-model-and-v2-call-using-seperate-search-hubs-in-standalone-searchbox/7429)

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/quanticStandaloneSearchBox.js [hunk 1]
```diff
@@ -76,6 +72,24 @@ export default class QuanticStandaloneSearchBox extends NavigationMixin(
    * @defaultValue '/global-search/%40uri'
    */
   @api redirectUrl = '/global-search/%40uri';
+  /**
+   * The [search hub](https://docs.coveo.com/en/1342/) to use for this Standalone Search Box.
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/quanticStandaloneSearchBox.js:76]: @jpmarceau could you please review these strings? Thanks!

**Review by jpmarceau — APPROVED** [review]
Strings look great, I have nothing to say 🥇 👌

**Comment by olamothe** [comment]
@nathanlb : This might interest you in relation to this PR

https://github.com/coveo/ui-kit/pull/1797

**Comment by nathanlb** [comment]
> @nathanlb : This might interest you in relation to this PR
> 
> #1797

Oh yes that is interesting. Thanks for pointing it out!

**Comment by erocheleau** [comment]
@nathanlb What happens if I set the same engineId on both the SSB and the search page component?
Whose searchhub/pipeline wins?
Is there any point in getting the searchhub from the data what was maybe specified in the Apex token generation class?

**Comment by nathanlb** [comment]
> @nathanlb What happens if I set the same engineId on both the SSB and the search page component? Whose searchhub/pipeline wins? Is there any point in getting the searchhub from the data what was maybe specified in the Apex token generation class?

The engineId given to the SSB gets appended with `_standalone` before being used in the SSB's dedicated search interface so you'd really have to want to do it for them to have the same ID. We did that to make the whole redirection work so they're supposed to be two distinct engines. If a user did make both IDs the same on purposeI believe the searchhub and pipeline values given to the SSB would take precedence but I also think the whole thing would work funky or not at all. I would say that's undefined and unsupported territory there.

If there are conflicts between values specified and the markup and the given JWT then you can see in Olivier's linked PR that it would be highlighted.

Does that answer your question?

File: packages/quantic/cypress/integration/standalone-search-box/standalone-search-box-expectations.ts [hunk 2]
```diff
@@ -75,6 +75,13 @@ function standaloneSearchBoxExpectations(
         })
         .logDetail('should log the "searchFromLink" UA event');
     },
+    fetchQuerySuggestWithParams: (
+      params: Record<string, string | number | boolean>,
+      alias: string
+    ) => {
+      cy.wait(alias);
+      cy.logDetail(`should fetch query suggestions with ${params}`);
```
Comment (lbergeron) [inline packages/quantic/cypress/integration/standalone-search-box/standalone-search-box-expectations.ts:83]: You should serialize `params` if you want the content to be displayed in the log.

```suggestion
      cy.logDetail(`should fetch query suggestions with ${JSON.stringify(params)}`);
```

File: packages/quantic/cypress/page-objects/search.ts [hunk 3]
```diff
@@ -6,10 +6,18 @@ import {
   // eslint-disable-next-line node/no-unpublished-import
 } from 'cypress/types/net-stubbing';
 
+type RequestParams = Record<string, string | number | boolean | undefined>;
+
 function uaAlias(eventName: string) {
   return `@UA-${eventName}`;
 }
 
+function compare(superset: RequestParams, subset: RequestParams) {
```
Comment (lbergeron) [inline packages/quantic/cypress/page-objects/search.ts:15]: I guess `includes` might be a better name for this function. Typically, `compare` refers to the equality operator.

**Review by lbergeron — APPROVED** [review]
It looks pretty nice!

**Comment by nathanlb** [comment]
> Nice 👌 Would it make sense to add the two new params in our [exampleStandaloneSearchbox.html](https://github.com/coveo/ui-kit/blob/91302445dcce770cfff43598d32a69e5984e7e29/packages/quantic/force-app/main/default/lwc/exampleStandaloneSearchbox/exampleStandaloneSearchbox.html#L6-L10) ?

Thanks! Technically yes but frankly I think we should remove that component. It isn't really an example component the same way the exampleSearch is as it isn't a recommended way of using it. It only serves to try the SSB in a localdevserver setting so I don't think we should be packaging it tbh. @erocheleau I think you're the one who added it a while ago. What do you think about removing it?

**Review by erocheleau — APPROVED** [review]

**Pull request merged**